src/fql.ts

```diff
@@ -221,19 +221,48 @@ function baseEvalFQL(fql: string, q: typeof query) {
   return fql.match(/^\s*{(.*\n*)*}\s*$/) ? eval(`(${fql})`) : eval(fql);
 }
 
-function splitQueries(code: string): string[] {
-  return code
-    .split(new RegExp(/(?:;|\n)/g))
-    .filter((item) => item.trim().length > 0)
-    .map((item) => item.trim());
+
+function parseQueries(code: string): string[] {
+  const brackets: Record<string, string> = {
+      '{': '}',
+      '(': ')',
+      '[': ']',
+      '"': '"',
+      '\'': '\'',
+  }
+  const openBrackets = new Set(Object.keys(brackets))
+  const closeBrackets = new Set(Object.values(brackets))
+  const queries = [];
+  const stack = [];
+  let start = 0;
+  let isOpening;
+  code = code.trim();
+
+  for (let i = 0; i < code.length; i++) {
+      if(openBrackets.has(code[i])){
+          stack.push(code[i]);
+          isOpening = true;
+      }
+
+      if(closeBrackets.has(code[i]) && brackets[stack.pop()!] !== code[i]){
+          throw new Error(`Unexpected closing bracket ${code[i]} at position: ${i + 1}`)
+      }
+
+      if(stack.length === 0 && isOpening) {
+        queries.push(code.slice(start, i + 1));
+          start = i + 1;
+          isOpening = false;
+      }
+  }
+  return queries;
 }
 
 export function runFQLQuery(code: string, client: Client) {
-  if (!code.trim()) {
-    return Promise.reject("Can not eval empty query.");
-  }
   try {
-    const queriesArray = splitQueries(code)
+    const queriesArray = parseQueries(code)
+    if(queriesArray.length ===0){
```

Comment:
Just a small one, there should be a space after the "===", and I notice there are spaces after "if"s as well.
Don't think it should be corrected here, but in my opinion, it would be nice to have some style checker as a separate task/PR. No need to it right now though.